plugins/modules/packaging/language/ansible_galaxy_install.py

```diff
@@ -0,0 +1,316 @@
+    - "B(Ansible 2.9): Returns empty because C(ansible-galaxy) has no C(list) subcommand."
+    type: dict
+    returned: always when installing roles
+    contains:
+      "<path>":
+        description: Roles and versions for that path.
+        type: dict
+    sample:
+      /home/user42/.ansible/roles:
+        ansistrano.deploy: 3.9.0
+        baztian.xfce: v0.0.3
+      /custom/ansible/roles:
+        ansistrano.deploy: 3.8.0
+  installed_collections:
+    description:
+    - If I(requirements_file) is specified instead, returns dictionary with all the collections installed per path.
+    - If I(name) is specified, returns that collection name and the version installed per path.
+    - "B(Ansible 2.9): Returns empty because C(ansible-galaxy) has no C(list) subcommand."
+    type: dict
+    returned: always when installing collections
+    contains:
+      "<path>":
+        description: Collections and versions for that path
+        type: dict
+    sample:
+      /home/az/.ansible/collections/ansible_collections:
+        community.docker: 1.6.0
+        community.general: 3.0.2
+      /custom/ansible/ansible_collections:
+        community.general: 3.1.0
+  new_collections:
+    description: New collections installed by this module.
+    returned: success
+    type: dict
+    sample:
+      community.general: 3.1.0
+      community.docker: 1.6.1
+  new_roles:
+    description: New roles installed by this module.
+    returned: success
+    type: dict
+    sample:
+      ansistrano.deploy: 3.8.0
+      baztian.xfce: v0.0.3
+"""
+
+import re
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import CmdModuleHelper, ArgFormat
+
+
+class AnsibleGalaxyInstall(CmdModuleHelper):
+    _RE_GALAXY_VERSION = re.compile(r'^ansible-galaxy(?: \[core)? (?P<version>\d+\.\d+\.\d+)(?:\.\w+)?(?:\])?')
+    _RE_LIST_PATH = re.compile(r'^# (?P<path>.*)$')
+    _RE_LIST_COLL = re.compile(r'^(?P<elem>\w+\.\w+)\s+(?P<version>[\d\.]+)\s*$')
+    _RE_LIST_ROLE = re.compile(r'^- (?P<elem>\w+\.\w+),\s+(?P<version>[\d\.]+)\s*$')
+    _RE_INSTALL_OUTPUT = None  # Set after determining ansible version, see __init_module__()
+    ansible_version = None
+    is_ansible29 = None
+
+    output_params = ('type', 'name', 'dest', 'requirements_file', 'force')
+    module = dict(
+        argument_spec=dict(
+            type=dict(type='str', choices=('collection', 'role', 'both'), required=True),
+            name=dict(type='str'),
+            requirements_file=dict(type='path'),
+            dest=dict(type='path'),
+            force=dict(type='bool', default=False),
+            ack_ansible29=dict(type='bool', default=False),
+        ),
+        mutually_exclusive=[('name', 'requirements_file')],
+        required_one_of=[('name', 'requirements_file')],
+        required_if=[('type', 'both', ['requirements_file'])],
+        supports_check_mode=False,
+    )
+
+    command = 'ansible-galaxy'
+    command_args_formats = dict(
```

Comment:
nit: the preferrable way of creating dicts in Python is to use literals like `{}`

Reply:
As preferrable ways go, I prefer this one because the editor paints the key and values in different colours, whereas using the json-like syntax strings are the same colour on both sides. Also, most of the modules I've seen in the collection use this syntax rather than the literal. And finally, I *trust* that when only constant keys and values are involved, the python compiler *should* be smart enough to generate a literal internally.